pages/sdk/signers/dynamic.mdx

```diff
@@ -0,0 +1,149 @@
+  <Tabs.Tab>
+    ```bash
+    npm install @dynamic-labs/sdk-react-core @dynamic-labs/ethereum
+    ```
+  </Tabs.Tab>
+  <Tabs.Tab>
+    ```bash
+    yarn add @dynamic-labs/sdk-react-core @dynamic-labs/ethereum
+    ```
+  </Tabs.Tab>
+  <Tabs.Tab>
+    ```bash
+    pnpm add @dynamic-labs/sdk-react-core @dynamic-labs/ethereum
+    ```
+  </Tabs.Tab>
+</Tabs>
+
+{/* <!-- vale on --> */}
+
+## Steps
+
+<Steps>
+
+  ### Imports
+
+  Here are the necessary imports for this guide.
+
+  {/* <!-- vale off --> */}
+
+  ```typescript
+  import { DynamicContextProvider, useDynamicContext, useIsLoggedIn } from '@dynamic-labs/sdk-react-core'
+  import { EthereumWalletConnectors, isEthereumWallet } from '@dynamic-labs/ethereum'
+  ```
+
+  {/* <!-- vale on --> */}
+
+  ### Initialize Dynamic
+
+  Dynamic works with React hooks. This means you can wrap your app with the `DynamicContextProvider` and have access to several react hooks like `useDynamicContext()` that will provide all the functionality.
+
+  `DynamicContextProvider` receives an `environmentId` and other configuration options. Check [Dynamic React SDK documentation](https://docs.dynamic.xyz/sdks/react-sdk/providers/dynamiccontextprovider) to learn more about all the different configuration options.
+
+{/* <!-- vale off --> */}
+
+  ```typescript
+  <DynamicContextProvider
+    settings={{
+      environmentId: YOUR_DYNAMIC_ENVIRONMENT_ID,
+      walletConnectors: [EthereumWalletConnectors],
+      // Add other configuration options as needed
+    }}
+  >
+    <App />
+  </DynamicContextProvider>
+  ```
+
+  {/* <!-- vale on --> */}
+
+  In this guide you will use the `primaryWallet` from the `useDynamicContext()` hook, and the `useIsLoggedIn()` hook.
+  
+  ```typescript
+  const { primaryWallet } = useDynamicContext()
+  const isLoggedIn = useIsLoggedIn()
+  ```
+
+  ### Login
+
+  In the App component, add [the DynamicWidget component](https://docs.dynamic.xyz/sdks/react-sdk/components/dynamicwidget) that will handle all the login and authentication logic for you in a best in class UI.
+
+{/* <!-- vale off --> */}
+
+  ```typescript
+  import { DynamicWidget } from '@dynamic-labs/sdk-react-core'
+
+  const App = () => {
+    return (
+      <>
+        // your other content here
+        <DynamicWidget />
+      </>
+    )
+  }
+  ```
+
+  {/* <!-- vale on --> */}
+
+  ### Get the provider and signer
+
+  Once the user is logged in through the DynamicWidget, you can get the `provider` and `signerAddress`, which is the externally owned account of the user that was derived from its credentials.
+  
+  To do that there is a `useEffect()` that executes when any of the `isLoggedIn` and `primaryWallet` variables have their values updated. Once they are all true, you have access to the user's connected signer. Note that we are also checking that the wallet is an Ethereum wallet using the `isEthereumWallet()` function.
+
+{/* <!-- vale off --> */}
+
+  <Tabs items={['viem']}>
```

Comment:
If `viem` is not used, the `Tabs` (including the import) should be removed.

Reply:
Done